### ISSUE TITLE
Multiple cases in switch statement

### ISSUE THREAD

**Issue by Remixful** [nhmkdev/cardmaker#99]
Could you add multiple cases to the switch statement?
For example:
`#(switch;@[type];(weapon;potion;equipment);item;default:@[type])#`
 It would really help.

Btw, this software has been fun (and easy) to use and learn so far. Thank you so much for taking the time to develop and share it with the rest of the game design community.

**Comment by nhmkdev** [comment]
I'll check into it. I have to be a bit cautious with all the delimiters that already exist in the translator logic!

As an alternative (available now) I would suggest using the "grouped if statement" for this logic. Sadly the manual has a terrible and incorrect description. It was functionality I was considering dropping but it's kind of useful.

I believe this would work for your situation (not quite as nice as a switch with overlapping cases of course!):
```
#(if @[type] == [weapon;potion;equipment] then item else @[type])#
```
You'll notice the @[type] doesn't have a `[]` around it but it could.

Basically this just checks that there is at least 1 item that matches between both sides of the `==`. So `[1;2;3] == [3;4;5]` would match for example.

Best of luck!

**Comment by Remixful** [comment]
That's exactly what I was looking for. Awesome, thanks! 😄

**Comment by Remixful** [comment]
It'll do for now. Thanks! 😄

**Comment by Remixful** [comment]
I actually am looking to use multiple cases, for example a different background for spells and creatures but the same background for items. I could make multiple elements and show/hide them individually, but it'd be nice if it could all fit into a line of code.

**Comment by Remixful** [comment]
I actually am looking to use multiple cases. I could make multiple elements and show/hide them individually, but it'd be nice if it could all fit into a line of code.

**Comment by nhmkdev** [comment]
The latest pre-release has switch support for groups of values in both the keys and cases now: [v.1.2.0.0-unstable.v.a8](https://github.com/nhmkdev/cardmaker/releases/tag/v.1.2.0.0-unstable.v.a8)

The sample below should work, just be sure to note that it uses a customized delimiter for the switch statement (the groups use ';').
```
#(switch::@[type]::[weapon;potion;equipment]::item::#default::@[type])#
```
